### ISSUE TITLE
database support?

### ISSUE THREAD

**Issue by gabrielcsapo** [apex/up#34]
Will there be an option for database support?

**Comment by tj** [comment]
Unlikely that I'll have anything special-cased to Up for the near-future, but later I could imagine having something to make setting up RDS and the others easier for people.

You'll of course be able to still fire up RDS, Compose, or whatever you prefer and use that. Did you have anything specific in mind?

**Comment by lukeed** [comment]
IMO this would be a separate-but-related product. A nice UI layer on top of AWS database options, with easy abstractions on deploy, replication, and backup/restore. 10/10

**Comment by tj** [comment]
If the metrics / dashboard stuff is written in a reasonably abstract way it should be pretty feasible to support all the AWS options, but definitely not a small task haha. Slippery slope there too, so many services, all vastly different, but we'll see!

**Comment by gabrielcsapo** [comment]
@tj thanks for the response, really cool product, which I could deploy it on a non AWS stack, something to the tune of a mesos cluster would be cool. I have been following this project since it was started and it reminded me of a much easier version of https://jujucharms.com, without the UI clumsiness

**Comment by vespertilian** [comment]
Having only used up for a few weeks I really enjoy it. It’s great for deploying the core of my app to Lambda, I chose this over the “Serverless” app project and am fairly happy with the results so far.  

I think some type of database support would be icing on the cake. 

I would just love to be able to go UP and then a whole app is deployed, including databases.

Based on the up marketing blurb, and this thread it looks like database deployment is not a focus, and I get that. 

I noticed that for Apex you mention Terraform to manage other infrastructure in the documentation. I have been looking into this and I feel like this could be a good avenue to pursue. Terraform or something else like Cloud Formation.

Currently you need to mess around with the permissions and make sure your Lambda function is connected to the right VPC. This is totally manageable, I managed it! I will hopefully even manage to get it sorted with Terraform. 

I just think there is an opportunity to make it a bit easier, for other Up users. Maybe instead of building out your own DB and infrastructure support you could point to some open source Terraform builds for common scenarios, built to work hand in hand with Up. 

To make some money here I was thinking you, or someone you bless could make a Udemy course that shows how to use Up alongside Terraform; and goes through deploying some standard apps with DB’s and caches etc. This might provide enough of a revenue stream to keep these terraform open source builds maintained. Personally I am a fan of video instruction, these videos alway make me more confident in adopting a new technology, so as a bonus a course like this might also help increase sales. 

I suggested a larger example app that is the sort of thing I would like to see in the video I suggested here: 

https://github.com/apex/up/issues/73

**Comment by tj** [comment]
@vespertilian agreed! I'd like to at least add the serverless solutions like Dynamo. Even Dynamo has a pretty large amount of configuration options so it might get weird, but I'll have to see how it feels, it would definitely be nice to have.

**Comment by kaihendry** [comment]
I'm using https://github.com/apex/gh-polls/blob/master/internal/poll/poll.go as a basis as data store for my App. It's working great.
